BitList.cs

```diff
@@ -0,0 +1,1553 @@
+        var intList = new List<int>();
+        foreach (var ints in intsList)
+        {
+            foreach (var num in ints)
+                intList.Add(num);
+        }
+        this.bits = new BitArray(intList.ToArray());
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A 16 bit character.</param>
+    public BitList(char bits)
+    {
+        this.bits = new BitArray(BitConverter.GetBytes(bits));
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A 16 bit character array (<see cref="System.Char"/> array).</param>
+    public BitList(char[] bits)
+    {
+        var bytesList = new List<byte[]>();
+        foreach (var num in bits)
+            bytesList.Add(BitConverter.GetBytes(num));
+        var byteList = new List<byte>();
+        foreach (var bytes in bytesList)
+        {
+            foreach (var num in bytes)
+                byteList.Add(num);
+        }
+        this.bits = new BitArray(byteList.ToArray());
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A 16 bit character array (<see cref="System.String"/>).</param>
+    public BitList(string bits)
+    {
+        var bytesList = new List<byte[]>();
+        foreach (var num in bits)
+            bytesList.Add(BitConverter.GetBytes(num));
+        var byteList = new List<byte>();
+        foreach (var bytes in bytesList)
+        {
+            foreach (var num in bytes)
+                byteList.Add(num);
+        }
+        this.bits = new BitArray(byteList.ToArray());
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A 16 bit character array of array.</param>
+    public BitList(string[] bits)
+    {
+        var _bytesList = new List<byte[]>();
+        var _byteList = new List<byte>();
+        foreach (var str in bits)
+        {
+            var bytesList = new List<byte[]>();
+            foreach (var chr in str)
+                bytesList.Add(BitConverter.GetBytes(chr));
+            var byteList = new List<byte>();
+            foreach (var bytes in bytesList)
+            {
+                foreach (var num in bytes)
+                    byteList.Add(num);
+            }
+            _bytesList.Add(byteList.ToArray());
+        }
+        foreach (var bytes in _bytesList)
+        {
+            foreach (var num in bytes)
+                _byteList.Add(num);
+        }
+        this.bits = new BitArray(_byteList.ToArray());
+    }
+
+    IEnumerator<bool> IEnumerable<bool>.GetEnumerator()
+    {
+        return new BitListEnumerator(this);
+    }
+
+    IEnumerator IEnumerable.GetEnumerator()
+    {
+        return new BitListEnumerator(this);
+    }
+
+    /// <summary>
+    /// Get the index of a bit in this instance.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    /// <returns>The <paramref name="bit"/> index.</returns>
+    public int IndexOf(bool bit)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename parameter 'bit' to 'item' to match the interface declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)